lib/transforms/input_legalizer.cc

```diff
@@ -112,6 +131,68 @@ bool InputLegalizer::RunOnFunction(Function* func) {
 
   HLCHECK(specified_shapes.empty() && "Unclaimed specified shapes");
 
+  if (!scale_str_.empty()) {
+    constexpr int n = 6;
+    auto v = ParsePreprocessScale(scale_str_);
+    if (func->Args().size() != 1) {
+      std::cerr << "Preprocess only works for model with exactly one input. "
+                   "Ignore preprocessing.\n";
+    } else if (v.size() != n) {
+      std::cerr << "Invalid number of scale parameters (" << v.size()
+                << " != " << n << ") Ignore preprocessing.\n";
+    } else {
+      // Insert scale.
+      auto arg = func->Args().begin()->get();
+      std::vector<float> bias(v.begin(), v.begin() + n / 2);
+      std::vector<float> scale(v.begin() + n / 2, v.end());
+      ConstantBuilder cb(func);
+      std::vector<int64_t> shape{n / 2};
+      const auto& arg_type = arg->GetResultType();
+      for (int64_t i = arg_type.GetNumOfDims() - 1; i >= 0; --i) {
+        if (arg_type.GetNumOfElementsInDim(i) != n / 2) {
+          shape.push_back(1);
+        }
+      }
+      // prepend ones.
+      for (int64_t i = 0, e = arg_type.GetNumOfDims() - shape.size(); i < e;
+           ++i) {
+        shape.insert(shape.begin(), 1);
+      }
```

Comment:
it may change the shape{1, n /2, 1, 1} to {1, 1, n/2, 1, 1}, should we add "break" in line 153 or rewrite it as follow?
``` 
size_t arg_dims = arg_type.GetNumOfDims();
std::vector<int64_t> shape(arg_dims, 1);
for (int64_t i = arg_dims - 1; i >= 0; --i) {
  if (arg_type.GetNumOfElementsInDim(i) == n / 2) {
    shape[i] = n / 2;
    break;
  }
}
```

Reply:
right. Let me re-work this part.